### PR TITLE
added homeWork_7

### PR DESCRIPTION
**Pull request by lesiaturuta**
Acceptance Criteria for 1

- [x]  Mentor is added to PR
- [x]  AC copied to PR description
- [x]  Link to deployed version is added to PR
- [x]  ProductPage fetches data from the API using either fetch or axios
- [x]  Pagination is implemented using query parameters
- [ ]  Search functionality is implemented using query parameters, with the search triggered by a search button
- [ ]  Category filtering is implemented using query parameters, resetting the page to the first page on change
- [ ]  A loading spinner or message is displayed while awaiting API requests
- [ ]  If the API response is an empty array or an error occurs, a "No products" message is displayed instead of a blank screen

link: https://lesiaturuta.github.io/fe-react-2024/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,63 +1,97 @@
...
 import type Product from '@/interface/product.ts';
 import getData from '@/utils/getData.ts';
 
 import './App.css';
 
+interface Category {
+    electronics: number;
+    shoes: number;
+    clothes: number;
+}
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: move to interfaces pls

File: src/App.tsx [hunk 2]
```diff
@@ -1,63 +1,97 @@
...
+    const [idCategory, setIdCategory] = useState<number>(0);
+    const [nameCategory, setNameCategory] = useState<string>('');
+    const [searchValue, setSearchValue] = useState<string>('');
+    const [isLoader, setIsLoader] = useState<boolean>(false);
+    const [errorMessage, setErrorMessage] = useState<string>('');
 
     const changeTheme = (name: string) => {
         name === 'light' ? setTheme('light') : setTheme('dark');
         localStorage.setItem('theme', name === 'light' ? 'light' : 'dark');
     };
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx:50]: maybe you can write more simple? it's hart to understand, better do refactoring for this.

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (lesiaturuta) [reply src/App.tsx]: fixed

File: src/App.tsx (hunk 2 above)
Comment (lesiaturuta) [reply src/App.tsx:50]: fixed

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**